lab_1_classify_by_unigrams/main.py

```diff
@@ -49,7 +87,24 @@ def compare_profiles(
     :param profile_to_compare: a dictionary of a profile to compare the unknown profile to
     :return: the distance between the profiles
     """
-
+    if not isinstance(unknown_profile, dict) or not isinstance(profile_to_compare, dict):
+        return None
+    if 'name' not in unknown_profile or 'freq' not in unknown_profile:
+        return None
+    if 'name' not in profile_to_compare or 'freq' not in profile_to_compare:
+        return None
+    if not isinstance(unknown_profile['freq'], dict) or not isinstance(profile_to_compare['freq'], dict):
+        return None
+    for i in unknown_profile['freq']:
+        if i not in profile_to_compare['freq']:
+            profile_to_compare['freq'][i] = 0
+    for j in profile_to_compare['freq']:
+        if j not in unknown_profile['freq']:
+            unknown_profile['freq'][j] = 0
+    unknown_profile['freq'] = dict(sorted(unknown_profile['freq'].items()))
+    profile_to_compare['freq'] = dict(sorted(profile_to_compare['freq'].items()))
```

Comment:
It is not a good practice to sort dictionaries as traditionally they are thought of as data structures that do not support specific order. If you want to use a dictionary that guarantees a specific order, you can look into Ordered Dictionaries, but I would recommend finding another workaround.

Reply:
I think that isalpha() is gong to check the symbol '\\' separately from letter. Isn't it true?

Reply:
How can I change sorted()? We need to have ordered lists to calculate mse, don't we?

Reply:
True, but dictionaries do not guarantee you order. Lists and tuples do. The problem is not in the `sorted`, it's in the fact that you use dictionaries to store your sorted values.

Reply:
Still a very important issue!